File: leetcode/go/190.reverse-bits.go
/*
 * @lc app=leetcode id=190 lang=golang
 *
 * [190] Reverse Bits
 */
package main

import (
	"fmt"
)

// @lc code=start
func reverseBits(num uint32) uint32 {
	var ret uint32
	var last uint32
	ret = 0
	total := 31
	for total >= 0 {
		if num > 0 {
			last = 0
			last = num & 1
			ret = (ret) ^ (last << total)
			num = num >> 1

		}
		total--
	}
	return ret
}

func main() {
	fmt.Println(reverseBits(43261596))
}

// @lc code=end

File: leetcode/cpp/190.reverse-bits.cpp
/*
 * @lc app=leetcode id=190 lang=cpp
 *
 * [190] Reverse Bits
 */
#include <iostream>
// @lc code=start
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int i = 0;
         uint32_t ret = 0;
        while(i < 31){
            if(n & 0x1){
                ret = ret + 1;
            }
            n = n >> 1;
            ret = ret << 1;
            ++i;
        }
        if(n & 0x1){
                ret = ret + 1;
            }
        return ret;
    }
};
// @lc code=end

